libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -356,10 +374,14 @@ AP_GPS::update_instance(uint8_t instance)
     uint32_t tnow = AP_HAL::millis();
 
     // if we did not get a message, and the idle timer of 1.2 seconds
-    // has expired, re-initialise the GPS. This will cause GPS
-    // detection to run again
+    // has expired, re-initialise the GPS. This will be delayed until
+    // after the current time is greater then the response_time_ms
+    // that is used by the driver to indicate that a long action may
+    // be underway until this time, and the driver should bot be unloaded.
+    // Unloading the driver will cause GPS detection to run again
     if (!result) {
-        if (tnow - timing[instance].last_message_time_ms > 1200) {
+        if (tnow - timing[instance].last_message_time_ms > 1200 &&
+            tnow > state[instance].response_time_ms) {
```

Comment:
I'm inclined to say no? The problem with resetting it then is that it is possible to have more data in the input buffer, which could cause the response time to reset to early. (It's possible to send the a slow message and receive a slow packet in the same call).

I've reduced the dependency on this so that only saving the config for the ublox needs this.


Neither @tridge or I actually like the implication that the GPS will go unresponsive while saving config for some of the u-blox's. It's only in at the moment because it's defaulted to off. Might be able to remove it with the rest of this patch, as now we can guarantee having set the configuration without needing to rely on just saving every time until we get it correctly.